Add BL0940 documentation (#1298)
OttoWinter/esphomedocs: components/sensor/bl0940.rst

Belling BL0940 Energy Monitor
==================================

.. seo::
    :description: Instructions for setting up BL0940 power monitors.
    :image: bl0940.png
    :keywords: BL0940

.. note::

    This page is incomplete and could some work. If you want to contribute, please read the
    :doc:`contributing guide </guides/contributing>`. This page is missing:

      - An image for the front page.

>       - Images/screenshots/example configs of this device being used in action.

Reviewer: I don't think this note is necessary anymore.

The ``bl0940`` sensor platform allows you to use BL0940 energy monitors sensors with

> ESPHome. These are used in some Tuya-devices (e.g. the power metering BW-SHP10)

Reviewer: ```suggestion
ESPHome. These are used in some Tuya-devices (e.g. the power metering BW-SHP10).
```

You must therefore have a ``uart:`` entry in your configuration with both the TX and RX pins set
to some pins on your board and the baud rate set to 4800 with 1 stop bit.

.. code-block:: yaml

    # Example configuration entry
    uart:
      id: uart_bus
      tx_pin: TX
      rx_pin: RX
      baud_rate: 4800
      stop_bits: 1

    sensor:
      - platform: bl0940
        uart_id: uart_bus
        voltage:
          name: 'BL0940 Voltage'
        current:
          name: 'BL0940 Current'
        power:
          name: 'BL0940 Power'
        energy:
          name: 'BL0940 Energy'
        temperature:
          name: 'BL0940 Internal temperature'
        target_temperature:
          name: 'BL0940 External temperature'
        update_interval: 60s

Configuration variables:
------------------------

- **voltage** (*Optional*): The voltage value of the sensor in Volts.
  All options from :ref:`Sensor <config-sensor>`.
- **current** (*Optional*): The current value of the sensor in Amperes. All options from
  :ref:`Sensor <config-sensor>`.
- **power** (*Optional*): The (active) power value of the sensor in Watts. All options from
  :ref:`Sensor <config-sensor>`.

> - **energy** (*Optional*): Use the voltage value of the sensor in kWh.
>   All options from :ref:`Sensor <config-sensor>`.

Reviewer: ```suggestion
- **energy** (*Optional*): The cumulative energy measured by the sensor in kWh.
  All options from :ref:`Sensor <config-sensor>`.
```
